src/main/com/yetanalytics/lrs/pedestal/http/multipart_mixed.cljc

```diff
@@ -18,58 +21,123 @@
           (mapv cs/trim
                 (cs/split h-str #"\s*:\s*" 2)))))
 
-;; TODO: Test!!!
-(defn- make-boundary-pattern
+;; A la https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html
+
+;; The requirement that the encapsulation boundary begins with a CRLF implies
+;; that the body of a multipart entity must itself begin with a CRLF before the
+;; first encapsulation line -- that is, if the "preamble" area is not used, the
+;; entity headers must be followed by TWO CRLFs. This is indeed how such
+;; entities should be composed. A tolerant mail reading program, however, may
+;; interpret a body of type multipart that begins with an encapsulation line NOT
+;; initiated by a CRLF as also being an encapsulation boundary, but a compliant
+;; mail sending program must not generate such entities.
+(defn- boundary-pat-open
+  [boundary]
+  (format "^(\\r\\n)?--%s\\r\\n" boundary))
+
+(defn- boundary-pat-mid
+  [boundary]
+  (format "(?<!^)\\r\\n--%s\\r\\n(?!$)" boundary))
+
+(defn- boundary-pat-close
   [boundary]
-  (re-pattern (str "(?m)\\R?^--" boundary "(?:--)?$\\R?")))
+  (format "\\r\\n--%s--(\r\n|\n)*$" boundary))
+
+(defn- assert-valid
+  [test message type-k]
+  (when-not test
+    (throw (ex-info message
+                    {:type type-k}))))
+
+#?(:cljs
+   (defn split-multiparts
+     "Splits multipart body parts, ensuring start + end and at least 2 parts"
+     [boundary body]
+     (let [[[open-idx open-bound]
+            :as open-re-pos] (u/re-pos
+                              (re-pattern
+                               (boundary-pat-open boundary))
+                              body)
+           mid-re-pos (u/re-pos
+                       (re-pattern
+                        (boundary-pat-mid boundary))
+                       body)
+           [[close-idx close-bound]
```

Comment:
`open-idx`, `open-bound`, `close-idx`, and `close-bound` are unused variables here, thanks to your recent commits deleting asserts. Which calls into question whether you need `u/re-pos` in the first place, as opposed to just `re-matches`.

Reply:
`re-matches` uses a single regex. I found that, rather than mucking about with a single giant one, I could be more explicit and maintain better control with 3:

Reply:
1. Anchored opening boundary regex

Reply:
Actually now that I think of it, it might be useful to use `open-idx` and `close-idx` in a _regular_ assert, for the purposes of sanity checking.

Reply:
(keep in mind the `:elide-asserts` compiler option is `false` by default on cljs

Reply:
https://github.com/yetanalytics/lrs/pull/83/commits/a50f6937aadd878aeed450d4286afd52cda2ff36